vendor/ocaml-4.12/location.ml

```diff
@@ -0,0 +1,956 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*             Xavier Leroy, projet Cristal, INRIA Rocquencourt           *)
+(*                                                                        *)
+(*   Copyright 1996 Institut National de Recherche en Informatique et     *)
+(*     en Automatique.                                                    *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+module List = struct
+  include List
+
+  (* Since OCaml 4.10 *)
+  let rec find_map f = function
+    | [] -> None
+    | x :: l ->
+        begin match f x with
+        | Some _ as result -> result
+        | None -> find_map f l
+        end
+end
```

Comment:
I would prefer if this was implemented outside of this module to keep the vendoring a bit reproducible.
For example, into a module `Parser_shim` that is passed with `-open`. This trick can help to replace the modules we do not care about (`Warning`, `Terminfo`, etc..) with dummy implementations that do nothing.

Ideally, vendored modules would be unchanged and every compatibility tweaks would be in a separate module. It will help a lot the future upgrades that these files can be overwritten easily.

Reply:
Good point, done!